.github/workflows/test.yml

```
@@ -15,7 +15,7 @@ jobs:
       - uses: ruby/setup-ruby@v1
         with:
           bundler-cache: true
-      - run: sudo npm install --global npm@7
+      - run: sudo npm install --global npm@8
```

Maybe, the following may work:

```suggestion
      - run: |
          echo "::set-output name=node::$(jq --raw-output '.engines.node' package.json)"
          echo "::set-output name=npm::$(jq --raw-output '.engines.npm' package.json)"
        id: node-versions
      - uses: actions/setup-node@v2
        with:
          node-version: ${{ steps.node-versions.outputs.node }}
      - run: npm install --global "npm@${{ steps.node-versions.outputs.npm }}"
```

Note that `sudo` is no longer needed.

---

Using [actions/setup-node](https://github.com/actions/setup-node)